src/modules/Api.ts

```diff
@@ -183,6 +190,89 @@ export class Api {
     });
   }
 
+  public async canonicalOrder({
```

Comment:
follow the naming conventions/structures of the previous functions (except we don't need a separate function for replaceOrder as `cancelId` is an optional field here

Reply:
this function name should be like `placeCanonicalOrder` or something